graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java

```diff
@@ -167,10 +167,27 @@ public void sourceMessagesForEvent(Event event, Consumer<List<MessageSummary>> m
                 }
                 messageConsumer.accept(summaries);
             };
+
+            ElasticsearchQueryString scrollQueryString = ElasticsearchQueryString.builder().queryString(config.query()).build();
+            scrollQueryString = scrollQueryString.concatenate(groupByQueryString(event));
+            LOG.debug("scrollQueryString: {}", scrollQueryString);
+
             final TimeRange timeRange = AbsoluteRange.create(event.getTimerangeStart(), event.getTimerangeEnd());
-            moreSearch.scrollQuery(config.query(), config.streams(), config.queryParameters(), timeRange, Math.min(500, Ints.saturatedCast(limit)), callback);
+            moreSearch.scrollQuery(scrollQueryString.queryString(), config.streams(), config.queryParameters(), timeRange, Math.min(500, Ints.saturatedCast(limit)), callback);
         }
+    }
 
+    // Return the ES query string for the group by fields specified in event; or empty if none specified
+    private ElasticsearchQueryString groupByQueryString(Event event) {
+        ElasticsearchQueryString result = ElasticsearchQueryString.empty();
+        if (!config.groupBy().isEmpty()) {
+            for (String key : event.getGroupByFields().keySet()) {
+                String value = event.getGroupByFields().get(key);
+                String query = key + ":" + value;
```

Comment:
This would fail for values with whitespaces or special lucene characters.
We would need to build the query wrapped in quotes:  `key:"my value"`
In the frontend we currently use this:
https://github.com/Graylog2/graylog2-server/blob/master/graylog2-web-interface/src/views/logic/queries/QueryHelper.ts

I think we could add a small helper class and use the lucene code from here:
(including their copyright, of course)
https://github.com/apache/lucene/blob/main/lucene/queryparser/src/java/org/apache/lucene/queryparser/classic/QueryParserBase.java#L954